dynamodb/schema.go

```diff
@@ -123,18 +123,31 @@ func analyzeMetadata(client dynamoClient, s *schema.Table) error {
 		return fmt.Errorf("failed to make a DescribeTable API call for table %v: %v", s.Name, err)
 	}
 
-	// Primary keys
+	// Primary keys.
 	for _, i := range result.Table.KeySchema {
 		s.PrimaryKeys = append(s.PrimaryKeys, schema.Key{Column: *i.AttributeName})
 	}
 
-	// Secondary indexes
+	// DynamoDB supports 2 types of indexes: Global Secondary Indexes (GSI) and Local Secondary Indexes (LSI).
+	// In GSI, dydb creates another global table for indexes that scales seperately.
+	// As for LSI, every partition in dydb maintains its own local index for that partition.
+	// For spanner, we should convert both these types as how dydb implements them is irrelevant.
+
```

Comment:
@bharadwaj-aditya please review if this explanation is sufficient.

Reply:
This looks fine. Also add the link to the doc for details - https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/LSI.html